src/eventmaster/plugins.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+var Plugins = map[string]Plugin{
+	"github": &GitHubPlugin{},
+}
+
+type Plugin interface {
+	ParseRequest(r *http.Request) (*UnaddedEvent, error)
+}
+
+type GitHubPlugin struct{}
+
+func (g *GitHubPlugin) ParseRequest(r *http.Request) (*UnaddedEvent, error) {
+	headers := r.Header
+	var tags []string
+	if eventType, ok := headers["X-Github-Event"]; ok {
+		tags = eventType
+	}
+	rawData, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var info map[string]interface{}
```

Comment:
It's not always guaranteed that "repo" and "pusher" info will exist since the contents of the data depend on the type of event - by creating a struct, would we try deserializing the data into the struct and if it fails, don't include that info?